src/Symfony/Component/RateLimiter/Policy/FixedWindowLimiter.php

```diff
@@ -70,8 +70,7 @@ public function reserve(int $tokens = 1, float $maxTime = null): Reservation
 
                 $reservation = new Reservation($now, new RateLimit($window->getAvailableTokens($now), \DateTimeImmutable::createFromFormat('U', floor($now)), true, $this->limit));
             } else {
-                $remainingTokens = $tokens - $availableTokens;
-                $waitDuration = $window->calculateTimeForTokens($remainingTokens);
+                $waitDuration = $this->interval;
```

Comment:
It seams like the issue is that `calculateTimeForTokens()` is looking at the "remaining tokens" while it should look for the total number of tokens the user wants to reserve. 

Would the following fix also work for you?
```suggestion
                $waitDuration = $window->calculateTimeForTokens($tokens);
```